version: "3.9"
services:
  db:
    image: mysql:8-oracle # -oracle tag supports arm64 architecture!
    restart: unless-stopped
    container_name: mlflow_db
    expose:
      - "3306"
    env_file:
      - /opt/mlflow/credentials.env
    volumes:
      - db_new_volume:/var/lib/mysql
    networks:
      - internal
  mlflow:
    image: ubuntu/mlflow:2.1.1_1.0-22.04
    container_name: tracker_mlflow
    restart: unless-stopped
    ports:
      - "5000:5000"
    env_file:
      - /opt/mlflow/credentials.env
    networks:
      - public
      - internal
    entrypoint: mlflow server --backend-store-uri mysql+pymysql://${MYSQL_USER}:${MYSQL_PASSWORD}@db:3306/${MYSQL_DATABASE} --default-artifact-root s3://${AWS_BUCKET_NAME}/ --artifacts-destination s3://${AWS_BUCKET_NAME}/ -h 0.0.0.0
    depends_on:
      wait-for-db:
        condition: service_completed_successfully
  wait-for-db:
    image: atkrad/wait4x
    depends_on:
      - db
    command: tcp db:3306 -t 90s -i 250ms
    networks:
      - internal
  run_test_experiment:
    build:
      context: ./test_experiment
      dockerfile: Dockerfile
    platform: linux/amd64  # once continuumio/miniconda3:latest image work on native aarch64 (arm), remove this line 
    depends_on:
      - "mlflow"
    environment:
      - MLFLOW_TRACKING_URI=http://mlflow:5000
    env_file:
      - /opt/mlflow/credentials.env
    entrypoint: >
      /bin/sh -c "
      python3 mlflow_tracking.py;
      exit 0;
      "
    networks:
      - internal
networks:
  internal:
  public:
    driver: bridge
volumes:
  db_new_volume:

